### ISSUE TITLE
[Screen Reader – BotKit - Docs]: Narrator is not narrating the Same Page link for multiple links in the Docs page.

### ISSUE THREAD

**Issue by v-gasikk** [howdyai/botkit-website#4]
**User Experience:**
Users who rely on Screen Readers will affect to navigate if Narrator is not narrating the Same Page link for multiple links in the page.

**Test Environment:**
OS: Win 10 Version 1809 (OS Build 17763.316)
URL: https://botkit.ai/docs/
Browser: Edge
Screen Reader: Narrator

**Repro Steps:**
Step 1: Open the above URL in Edge browser.
Step 2: Navigate to “Docs” link and select it.
Step 3: Navigate to all header links in the page.
Step 4: Verify whether Narrator is narrating the Same Page link for the links in the page.

**Actual Result:**
Narrator is not narrating the Same Page link for below links in the page.
Botkit: Building Blocks for Building Bots
Install BotKit
Remix on Glitch
Command Line Interface
Start from scratch
Build your Bot
Hearing Messages
Responding to Events
Conversation Management
Extend Botkit with Middleware
What’s Next

Ex: When focus is on "Botkit: Building Blocks for Building Bots" link, Narrator is narrating as "Botkit: Building Blocks for Building Bots" link. If we click on the link it is opening in same page.

**Expected Result:**
Narrator should narrate the Same Page link for below links in the page.
Botkit: Building Blocks for Building Bots
Install BotKit
Remix on Glitch
Command Line Interface
Start from scratch
Build your Bot
Hearing Messages
Responding to Events
Conversation Management
Extend Botkit with Middleware
What’s Next

Ex: When focus is on "Botkit: Building Blocks for Building Bots" link, Narrator should narrate "Botkit: Building Blocks for Building Bots" same page link.

**Note:** Visually above links are not looking like as links. 
**Note**: Same issue repros in all pages which are present under “Docs” link.

**Note**: Same issue repros in Chrome + NVDA also.

**MAS Reference Link:**
[](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={06a891af-210c-4c6d-bb3e-92babab209e7})
![MAS2 4 4_BotKit_Docs_Same Page Link](https://user-images.githubusercontent.com/48272005/54332570-c2b6ef80-4644-11e9-99f1-3d948b81b00b.png)
[MAS2.4.4_BotKit_Docs_Same Page Link_Video.zip](https://github.com/howdyai/botkit-website/files/2964779/MAS2.4.4_BotKit_Docs_Same.Page.Link_Video.zip)

**Comment by benbrown** [comment]
@awalia13  How do I specify that a link is a same page link?

**Comment by benbrown** [comment]
Regarding this:

> Note: Visually above links are not looking like as links.

These really are HEADERS, which have been made into links so that developers can deep-link to the specific section of the documentation.  I do not want them to appear underlined like a link.  Are there acceptable alternative ways to indicate this?

**Comment by benbrown** [comment]
after discussion with @awalia13, we will leave header links alone.

**Comment by benbrown** [comment]
I have added the aria-current="page" attribute to each page, so hopefully the "Same page" narration is now fixed.

**Comment by v-gasikk** [comment]
Verified. Issue is Not Fixed. Hence Re-activating the bug.
aria-current="page" attribute is not added for all links in the page. Narrator is narrating as link when focus is on header same page links.
![MAS2 4 4_Not Fixed](https://user-images.githubusercontent.com/48272005/55381636-3a3fb680-5541-11e9-9f03-32bc14e56f70.png)

**Comment by benbrown** [comment]
@v-gasikk @awalia13 Are you saying that _every anchor_ that points to the same page should have aria-current="page" on it?

**Comment by v-gasikk** [comment]
Yes

**Comment by benbrown** [comment]
@v-gasikk all of these links have now been tagged with `aria-current="page"` please revalidate

**Comment by v-gasikk** [comment]
Verified. Issue is Not Fixed. Hence Re-activating the bug.
aria-current="page" attribute is not added for all links in the page. Narrator is narrating as link when focus is on header same page links.
![Not Fixed same page link](https://user-images.githubusercontent.com/48272005/56112650-539d2580-5f79-11e9-9141-67bb427fdbe7.png)

**Comment by benbrown** [comment]
Apologies - my change got inadvertently rolled back.

Please revalidate. Thanks.

**Comment by awalia13** [comment]
This is a Narrator bug as it is working fine with other combinations. Already logged a separate bug with Narrator.
